src/main/java/seedu/address/model/person/Person.java

```diff
@@ -104,71 +235,122 @@ public PersonBuild<T> tags(Set<Tag> tags) {
+        }
+
+        public Name getName() {
+            return this.name;
+        }
+
+        public Phone getPhone() {
+            return this.phone;
+        }
+
+        public Email getEmail() {
+            return this.email;
+        }
+
+        public Address getAddress() {
+            return this.address;
+        }
+
+        public Set<Tag> getTags() {
+            return this.tags;
+        }
+
+        public ObservableList<Person> getPairedPersons() {
+            return this.pairedPersons;
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            if (other == this) {
+                return true;
+            }
+
+            // instanceof handles nulls
+            if (!(other instanceof PersonBuilder)) {
+                return false;
+            }
+
+            PersonBuilder otherPersonBuilder = (PersonBuilder) other;
+            return Objects.equals(name, otherPersonBuilder.name)
+                    && Objects.equals(phone, otherPersonBuilder.phone)
+                    && Objects.equals(email, otherPersonBuilder.email)
+                    && Objects.equals(address, otherPersonBuilder.address)
+                    && Objects.equals(tags, otherPersonBuilder.tags)
+                    && Objects.equals(pairedPersons, otherPersonBuilder.pairedPersons);
+        }
+
+        @Override
+        public String toString() {
+            return new ToStringBuilder(this)
+                    .add("name", name)
+                    .add("phone", phone)
+                    .add("email", email)
+                    .add("address", address)
+                    .add("tags", tags)
+                    .add("paired persons", pairedPersons)
+                    .toString();
+        }
     }
 
     /**
      * Constructor for a Person object using the builder. This is the intended method of constructing
      * the person object via the Builder pattern.
      */
-    public Person(PersonBuild<? extends PersonBuild<?>> builder) {
-        requireAllNonNull(builder.name, builder.phone, builder.email, builder.address,
-                builder.tags, builder.pairedPersons);
+    public Person(PersonBuilder builder) {
+        requireAllNonNull(builder.name);
         this.name = builder.name;
-        this.phone = builder.phone;
-        this.email = builder.email;
-        this.address = builder.address;
-        this.tags = builder.tags;
-        this.pairedPersons = builder.pairedPersons;
+        this.phone = builder.phone != null ? builder.phone : DEFAULT_PHONE;
+        this.email = builder.email != null ? builder.email : DEFAULT_EMAIL;
+        this.address = builder.address != null ? builder.address : DEFAULT_ADDRESS;
+        this.tags = builder.tags != null ? builder.tags : new HashSet<>();
+        this.pairedPersons = builder.pairedPersons != null
+                ? builder.pairedPersons : FXCollections.observableArrayList();
```

Comment:
Very minor nit: Would it be clearer to use `Collections.emptySet()` and `FXCollections.emptyObservableList()` to make the intended meaning of the code clearer?

```suggestion
        this.tags = builder.tags != null ? builder.tags : Collections.emptySet();
        this.pairedPersons = builder.pairedPersons != null
                ? builder.pairedPersons : FXCollections.emptyObservableList();
```